src/OpenLoco/src/Vehicles/OrderManager.cpp

```diff
@@ -98,6 +110,38 @@ namespace OpenLoco::Vehicles::OrderManager
         }
     }
 
+    // 0x004705C0
+    void deleteOrder(VehicleHead* head, uint16_t orderOffset)
+    {
+        // Find out what type the selected order is
+        OrderRingView orders(head->orderTableOffset, orderOffset);
+        auto& selectedOrder = *(orders.begin());
+
+        auto removeOrderSize = kOrderSizes[enumValue(selectedOrder.getType())];
+        head->sizeOfOrderTable -= removeOrderSize;
+
+        // Are we removing an order that appears before the current order? Move back a bit
+        if (head->currentOrder > orderOffset)
+        {
+            head->currentOrder -= removeOrderSize;
+        }
+
+        // Ensure we don't move beyond the order table size
+        if (head->currentOrder + 1 >= head->sizeOfOrderTable)
+        {
+            head->currentOrder = 0;
+        }
+
+        // Move orders in the order table, effectively removing the order
+        shiftOrdersDown(head->orderTableOffset + orderOffset, removeOrderSize);
+
+        // Bookkeeping: change order table size
+        numOrders() -= removeOrderSize;
+
+        // Compensate other vehicles to use new table offsets
+        reoffsetVehicleOrderTables(head->orderTableOffset + orderOffset, -removeOrderSize);
```

Comment:
should be 1 increment past this that you reoffset otherwise you will double remove the current vehicles offset.

Reply:
Thanks for catching that. Addressing this fixes the remaining issue for me as well.